From 0b539ed33e60bf2a67f9e6d2600e34f858dac9d6 Mon Sep 17 00:00:00 2001
From: Yuan Tong <tongyuan.gen@bytedance.com>
Date: Tue, 5 Dec 2023 15:28:28 +0800
Subject: [PATCH 2/4] relax CXX standard requirement

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9e2129b..bd2d22f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -53,7 +53,7 @@ else()
     set(CMAKE_CXX_STANDARD 17)
 endif()
 
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_STANDARD_REQUIRED OFF)
 set(CMAKE_CXX_EXTENSIONS OFF)
 
 if (${COVERAGE})
