diff --git a/bmf/hml/src/kernel/image_iter.h b/bmf/hml/src/kernel/image_iter.h
index ba08aa9..9b2d9a3 100644
--- a/bmf/hml/src/kernel/image_iter.h
+++ b/bmf/hml/src/kernel/image_iter.h
@@ -446,7 +446,7 @@ struct YUVIter<
 
 #define HMP_CHANNEL_FORMAT_DISPATCH_CASE(Format, ...)                          \
     case (Format): {                                                           \
-        const auto FMT = Format;                                               \
+        constexpr auto FMT = Format;                                           \
         return __VA_ARGS__();                                                  \
     }
 
